src/interfaces/records/messages/records-write.ts

```diff
@@ -119,6 +119,10 @@ export class RecordsWrite extends Message<RecordsWriteMessage> {
     await validateAuthorizationIntegrity(message, { allowedProperties: new Set(['recordId', 'contextId', 'attestationCid', 'encryptionCid']) });
     await RecordsWrite.validateAttestationIntegrity(message);
 
+    if (message.descriptor.protocol !== undefined) {
+      validateProtocolUriNormalized(message.descriptor.protocol);
+    }
```

Comment:
Try putting the check within `validateIntegrity()`, that's the intent of that method. Not picking on other methods since the checks you introduced are the "first check", if we have more than 1 check in the future we'd follow the same pattern also.